### DISCUSSION TITLE
Gears with Tacx Flow Smart

### DISCUSSION THREAD

**Discussion by jukrauss** [cagnulein/qdomyos-zwift#1459] [Q&A] [answered]
Hey,
i just bought this app and need help to set it up.

You can find a description of my problem here:
https://forums.zwift.com/t/increase-baseline-resistance/19109/164?u=dragon_fox

I just want to be able to use lower gears on my bike with higher wattage, to reduce speed on my on wheel trainer. I think about a flat increase of everything zwift sends my trainer of about 30-50%. 

A user said i can achieve this with this app and Roberto Viola said i should open a thread here. 

Just as repetition of what was said in the zwift forum so far:
_Just bought it and fiddle around. It detects my trainer. My zwift, now detects it as Wahoo KICKR 0000 33. But i am not sure what settings i have to use.
I red the documentation on github and i restart between every test.
I tried Ziwft Resistance Offset, Gain and Rolling Resistance Factor. Nothing changes. Do i miss something?"_

Roberto Viola said i need the gear tile. 
I activated the gear tile, restarted and set gears between 0 and 25. No changes at all. 

Is there something else i need to do?

**Comment by cagnulein** [comment]
hi @jukrauss welcome here :)
Send me a debug log of you trying to increase the gear tile and I will check what it's happening to your setup. https://github.com/cagnulein/qdomyos-zwift/wiki/How-do-i-get-the-debug-log-in-case-something-doesn't-work%3F

Thanks

**Comment by jukrauss** [comment]
Well... I think i solved it. 

The problem sits on the bike.

I increased the gear in the app and decreased the gear on the bike. Hence... The effort kept almost the same.

However, is there a way to regulate it more granular? I have the feeling that for gear >3 the power stays the same and between 0 and 2 the differences are very big. Is there a way to soften that a little bit?


Here is my log. I am not sure if needet anymore or helpful in the first place. The first minutes i had connection problems and then i just played around with the gears and drove short distances with same gear and cadence to see what changes.

[debug-Sat_May_13_15_15_14_2023.log](https://github.com/cagnulein/qdomyos-zwift/files/11469770/debug-Sat_May_13_15_15_14_2023.log)

**Reply by cagnulein** [reply, marked as answer]
hi @jukrauss the debug log is very clear. Just some quick info in order to give you an overview how the gear tile is working.

When you change the gear tile, the mod is applied to the next inclination received from zwift. So this means that if you are on a flat road and you're trying to change the gears mod, this is not applied until the next uphill or downhill. Of course I can fix this (this is valid only for your trainer model, all the others trainers change immediately)

Also, how does it work the gear tile? In the current implementation, i'm adding a 1% grade step every time you push the gear tile. So for example, if you are on a 0% grade on Zwift, and you're setting the gear tile to 1, it means that i'm converting the 0% zwift grade to 1% grade to your trainer and so on.

So yes, actually the granularity is not so great but there is a good news: i can easily fix it! Maybe adding a new setting to set the granularity, what do you think?

So, summarizing, your requests are:
- change the virtual gear every time I'm pressing the gear tile immediately
- change the granularity

Is it ok for you?

**Comment by jukrauss** [comment]
Well... I didn't expect it to become a request in the first place. If I know how it works that is already a big help for me. 

For the first topic:
If it is not a big effort it would be great if gears would have an effect immediately. Then someone could simply evaluate the correct gear very quick on a long straight and leave it at that. 
So it's just a little more time consuming because you always have to be up to the next slope and then back on a flat piece for a reference. Having no immediate change is not intuitive and it even looks like the feature does not work at all.

So yeah. It is not realy a request for me personally but nice to have. I can go long ways for workarounds.


For the granularity:
It is good to know on how it works. I just have to test it a little bit more, which i can do later or tomorrow. I guess, so far i increased and decreased it most of the time without seeing changes because i did not know that i have to wait for a sloope. 

For the request: Yes. I think more granularity would be nice. It would allow to manually setup an even more realistic feeling. 
If you allow me a draft:
Maybe it is possible to have a multiplicative setting with minimum 0.1 and maximum 1 and default 1. So that the setting defines the percentage of increased slope from 0.1 to 1 per gear? That way, it does not break all the useres gear settings and would allow increasing and decreasing in different steps between 0.1% and 1% per gear. 


However: 
This is not, by any means, a necessary or game changing request for me. For now, i am a happy customer and user whose problem seems to be solved.
 
Thank you very much.

**Reply by cagnulein** [reply]
no problem for me! they are 2 very easy task! so I will be glad to do them! let me raise a new ticket for this! I will let you know when I will be ready to test
